File: server/java/src/main/java/com/onlinepayments/example/service/domain/payments/Mandate.java
package com.onlinepayments.example.service.domain.payments;

import com.onlinepayments.example.service.domain.common.Address;
import com.onlinepayments.example.service.domain.payments.enums.RecurrenceType;
import com.onlinepayments.example.service.domain.payments.enums.SignatureType;

public class Mandate {
    private String iban;
    private String customerReference;
    private String mandateReference;
    private RecurrenceType recurrenceType;
    private SignatureType signatureType;
    private String returnUrl;
    private Address address;

    public Mandate() {}
    public Mandate(String iban, String customerReference, String mandateReference,
                   RecurrenceType recurrenceType, SignatureType signatureType,
                   String returnUrl, Address address) {
        this.iban = iban;
        this.customerReference = customerReference;
        this.mandateReference = mandateReference;
        this.recurrenceType = recurrenceType;
        this.signatureType = signatureType;
        this.returnUrl = returnUrl;
        this.address = address;
    }
    public String getIban() { return iban; }
    public String getCustomerReference() { return customerReference; }
    public String getMandateReference() { return mandateReference; }
    public RecurrenceType getRecurrenceType() { return recurrenceType; }
    public SignatureType getSignatureType() { return signatureType; }
    public String getReturnUrl() { return returnUrl; }
    public Address getAddress() { return address; }

    public void setIban(String iban) {
        this.iban = iban;
    }
    public void setCustomerReference(String customerReference) {
        this.customerReference = customerReference;
    }
    public void setMandateReference(String mandateReference) {
        this.mandateReference = mandateReference;
    }
    public void setRecurrenceType(RecurrenceType recurrenceType) {
        this.recurrenceType = recurrenceType;
    }
    public void setAddress(Address address) {
        this.address = address;
    }
}

File: server/ruby/app/business/domain/payments/mandate.rb
module Business
  module Domain
    module Payments
      class Mandate
        attr_accessor :iban,
                      :customer_reference,
                      :mandate_reference,
                      :recurrence_type,
                      :signature_type,
                      :return_url,
                      :address

        def initialize(iban: nil,
                       customer_reference: nil,
                       mandate_reference: nil,
                       recurrence_type: nil,
                       signature_type: nil,
                       return_url: nil,
                       address: nil)
          @iban = iban
          @customer_reference = customer_reference
          @mandate_reference = mandate_reference
          @recurrence_type = recurrence_type
          @signature_type = signature_type
          @return_url = return_url
          @address = address
        end

        def self.from_sdk_hash(hash)
          return nil unless hash.is_a?(Hash)

          recurrence_type = Business::Domain::Enums::Payments::RecurrenceType.try_from(hash[:recurrence_type])
          signature_type = Business::Domain::Enums::Payments::SignatureType.try_from(hash[:signature_type])

          new(
            iban: hash[:iban],
            customer_reference: hash[:customer_reference],
            mandate_reference: hash[:mandate_reference],
            recurrence_type: recurrence_type,
            signature_type: signature_type,
            return_url: hash[:return_url],
            address: hash[:address] ? Business::Dtos::Common::Address.from_sdk_hash(hash[:address]) : nil
          )
        end

        def to_sdk_hash
          {
            'iban' => @iban,
            'customerReference' => @customer_reference,
            'mandateReference' => @mandate_reference,
            'recurrenceType' => @recurrence_type,
            'signatureType' => @signature_type,
            'returnUrl' => @return_url,
            'address' => @address&.to_h
          }
        end
      end
    end
  end
end